### PR TITLE
Add SASS prefix handler

### PR DESCRIPTION
**Pull request by JetyAdam**
Jira ticket: RHCLOUD-24774

### CODE REVIEW

File: packages/config/src/lib/createConfig.ts
Comment (Hyperkid123) [inline packages/config/src/lib/createConfig.ts:127]: I would create a constant and remove the `shouldPrependPrefix`

File: packages/config/src/lib/createConfig.ts
Comment (Hyperkid123) [inline packages/config/src/lib/createConfig.ts:129]: This can be removed once appName and sassPrefix is merged into one argument.

File: packages/config/src/lib/createConfig.ts
Comment (Hyperkid123) [inline packages/config/src/lib/createConfig.ts:136]: This matches even nested rules.

**Review by Hyperkid123 — COMMENTED** [review]
Could you extract te function to a separate file. Also write some tests for the input/output.

File: packages/config/src/lib/addPrefixToContent.ts
Comment (Hyperkid123) [inline packages/config/src/lib/addPrefixToContent.ts:1]: Do you need the TODO?

File: packages/config/src/lib/addPrefixToContent.test.ts
Comment (Hyperkid123) [inline packages/config/src/lib/addPrefixToContent.test.ts:1]: I'd also add a mixed example. A content with rules that should be prefixed and rules that should not be prefixed.

And mix up the order of the rules.

**Comment by Hyperkid123** [comment]
Add a custom matcher pealse to remove the replace for all the strings.

File: packages/config/src/lib/addPrefixToContent.ts
Comment (bastilian) [inline packages/config/src/lib/addPrefixToContent.ts:5]: Maybe instead of building a custom parser, we can utilise sass' `compileString` function? https://sass-lang.com/documentation/js-api/functions/compilestring/

Comment (bastilian) [reply packages/config/src/lib/addPrefixToContent.ts:5]: Hm. I thought maybe compileString would allow to have a more structured representation of the css, but it's just a string too.

Comment (Hyperkid123) [reply packages/config/src/lib/addPrefixToContent.ts:2]: Oh, we don't want to compile, we will leave that to saas. We just want to not prefix css rules that match the app name. To allow styling the root element. I think is a manual process. I wish there was some nice AST for scss that is not 10 years old.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**
